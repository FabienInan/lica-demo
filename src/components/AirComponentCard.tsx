import React from 'react';
import styled from 'styled-components';

const AirComponentCardContainer = styled.div`
    margin: 8px;
    height: 148px;
    width: 148px;
    border-radius: 8px;
    box-shadow: 0.3px 0.3px 0.3px 1px rgba(0, 0, 0, 0.2);
    text-align: center;
`;

const Name = styled.div`
    color: gray;
    line-height: 40px;
    font-size: 20px;
    height: 80px;
`;

const Concentration = styled.div`
    font-size: 22px;
`;

type AirComponentCardProps = {
    name: string;
    value: number;
};

const AirComponentCard = (props: AirComponentCardProps) => {
    return (
        <AirComponentCardContainer>
            <Name>{props.name}</Name>
            <Concentration>{props.value}Î¼g/m3</Concentration>
        </AirComponentCardContainer>
    )
};

export default AirComponentCard;