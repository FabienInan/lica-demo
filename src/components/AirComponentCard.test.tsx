import AirComponentCard from './AirComponentCard';
import React from 'react';
import renderer from 'react-test-renderer';

it('should render properly the Air Component Card', () => {
    const props = {
        name: 'Monoxyde de carbone',
        value: 12.35
    };
    const tree = renderer.create(<AirComponentCard name={props.name} value={props.value}/>).toJSON();
    expect(tree).toMatchSnapshot();
});