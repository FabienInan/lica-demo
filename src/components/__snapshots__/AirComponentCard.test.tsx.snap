// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly the Air Component Card 1`] = `
<div
  className="sc-bdvvtL kIatee"
>
  <div
    className="sc-gsDKAQ kuZoDU"
  >
    Monoxyde de carbone
  </div>
  <div
    className="sc-dkPtRN keSRXu"
  >
    12.35
    Î¼g/m3
  </div>
</div>
`;
